firewood/src/proof.rs

```diff
@@ -102,11 +102,95 @@ const BRANCH_NODE_SIZE: usize = 17;
 /// to a single proof step. If reaches an end step during proof verification,
 /// the hash value will be none, and the encoded value will be the value of the
 /// node.
-pub struct SubProof {
+struct SubProof {
     encoded: Vec<u8>,
     hash: Option<[u8; 32]>,
 }
 
+fn generate_subproof(data: Vec<u8>) -> Result<SubProof, ProofError> {
```

Comment:
nit: `SubProof::try_new(data)`

Reply:
Can do that in a follow-up. All I'm doing in this PR is moving the code that already exists since it didn't make sense the way it was laid out before.

Reply:
Sure. I usually do this as a separate commit -- move in the first commit, then rename and refactor next. If I see the function move then I can study the differences separately if I so desire.

Otherwise I think these smaller nit comments get lost (at least for me). They are all, however, nits so nothing stops merging here.